### DISCUSSION TITLE
how to typing page meta?

### DISCUSSION THREAD

**Discussion by hi-reeve** [nuxt/nuxt#19949] [Questions]
as per [docs](https://nuxt.com/docs/guide/directory-structure/pages/#typing-custom-metadata)
we need to define a new pageMeta module inside `index.d.ts`
i've done this, here is my `index.d.ts` file
 
```ts
import { Permission } from "./utils/permission";

declare module '#app' {
	interface PageMeta {
		auth?: boolean;
		isBack?: boolean;
		pageHeader?: boolean;
		customer?: boolean;
		admin?: boolean;
		permission?: Permission;
	}
}

export {}
``` 

but when i try to use `definePageMeta` it does not have the autocomplete.
<img width="463" alt="image" src="https://user-images.githubusercontent.com/45350572/228142942-ffef9310-08ad-4ad2-ae4f-02e9ebaf86ac.png">

what should i do? do i need to manually override `tsconfig.json` ?

**Comment by masyoudi** [comment]
same issue

**Comment by masyoudi** [comment]
i'm trying to add typing into bottom of nuxt.config.ts, and works fine

```ts
import type { Permissions } from '~~';

export default defineNuxtConfig({
  //
});

declare module '#app' {
  interface PageMeta {
    permission?: Permissions;
  }
}
```

**Reply by lieven121** [reply]
Still an issue

Thx for the workaround @masyoudi

**Comment by angelhdzmultimedia** [comment]
```ts 
declare module 'nuxt/app'
```
![image](https://github.com/user-attachments/assets/d6bf3594-e7da-49bd-b335-f497bf9af485)

**Edit:** I think the solution is restarting nuxt dev server, so it scan the source dir for the index.d.ts and update the typings for `PageMeta`.
The docs mention `declare module '#app` so it's safe to use that import alias.

Edit 2: You can add a nuxt module that injects the types as a virtual file, so that the file can be registered for HMR, and everytime you modify the file, it triggers a HMR render. https://nuxt.com/docs/guide/going-further/modules#adding-type-declarations
